### PR TITLE
Word-Break-Problem

### PR DESCRIPTION
**Pull request by Meghag5**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes.)

### Description

(This problem is solved using backtracking in which we divide single string with valid words.)
eg.
input - ilovecoding
output - i love coding

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#2295

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Few changes: 

- Don't make a new section `Backtracking`
- Add a beginner friendly pitch about the Code
- Add sample I/O at the end of the Code
- Add Code Complexity at the end of the Code
- Format your Code using a clang formatter 
- Use dynamic input from the user

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -219,6 +220,7 @@ _add list here_
 
 - [Convex Hull (Jarvis' Algorithm)](computational_geometry/Convex_Hull_Jarvis_Algorithm.cpp)
 
+
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Remove the extra line

File: C-Plus-Plus/README.md [hunk 2]
```diff
@@ -175,6 +175,7 @@ _add list here_
 - [Pythagorean Triples Generator](math/Pythagorean_Triples.cpp)
 - [Sieve of Eratosthenes](math/Sieve_of_Eratosthenes.cpp)
 - [Ugly Numbers](math/ugly_numbers.cpp)
+-[Word Break Problem](backtracking/Word_break.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Give a space between `-` and `[`

Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Why are you adding backtracking directory here? Didn't I tell you to remove that?

File: C-Plus-Plus/math/Word_break.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
+#include <bits/stdc++.h>
```
Comment (HarshCasper) [inline C-Plus-Plus/math/Word_break.cpp]: Add a beginner friendly pitch about the Code

File: C-Plus-Plus/math/Word_break.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    BreakWords("microsofthiring");
+    return 0;
+}
+
+//eg.
+//input - ilovecoding
+//output - i love coding
+
+//Time Complexity - O(n2)
+//Space Complexity - O(1)
```
Comment (HarshCasper) [inline C-Plus-Plus/math/Word_break.cpp]: Keep them in multi-line comments

File: C-Plus-Plus/math/Word_break.cpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+
+// It will Print all possible word breaks of given string
+void BreakWords(string str){
+    breakingwords(str,str.size(),"");
+}
+
+int main() {
+
+    BreakWords("microsofthiring");
```
Comment (HarshCasper) [inline C-Plus-Plus/math/Word_break.cpp]: Use dynamic input

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/README.md (hunk 2 above)
Comment (Meghag5) [reply C-Plus-Plus/README.md]: really sorry i removed backtracking directory but forgot to write math

**Comment by Meghag5** [comment]
pls review it again made all the changes asked @HarshCasper

File: C-Plus-Plus/math/Word_break.cpp [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+//eg.
+//input - microsofthiring
+//output - 
+// micro soft hi ring
+// micro soft hiring
+// microsoft hi ring
+// microsoft hiring
+//Time Complexity - O(n2)
+//Space Complexity - O(1)
```
Comment (HarshCasper) [inline C-Plus-Plus/math/Word_break.cpp]: Put in multi-line comments

File: C-Plus-Plus/math/Word_break.cpp [hunk 7]
```diff
@@ -0,0 +1,56 @@
+#include <iostream>
```
Comment (HarshCasper) [inline C-Plus-Plus/math/Word_break.cpp:2]: Add a beginner friendly description

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Format the Code properly

File: C-Plus-Plus/math/Word_break.cpp (hunk 6 above)
Comment (Meghag5) [reply C-Plus-Plus/math/Word_break.cpp]: done

File: C-Plus-Plus/math/Word_break.cpp (hunk 7 above)
Comment (Meghag5) [reply C-Plus-Plus/math/Word_break.cpp:2]: done

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**